test/modules/TestResultCoordinator/Reports/TestReportGeneratorFactory.cs

```diff
@@ -42,13 +47,20 @@ internal TestReportGeneratorFactory(ITestOperationResultStorage storage, Network
                         var expectedTestResults = this.GetResults(metadata.ExpectedSource);
                         var actualTestResults = this.GetResults(metadata.ActualSource);
 
+                        await this.LonghaulSettings.ForEachAsync(async (longhaulSettings) =>
+                        {
+                            TestResultFilter filter = new TestResultFilter(new SimpleTestOperationResultComparer());
+                            TimeSpan unmatchedResultTolerance = longhaulSettings.UnmatchedResultTolerance;
+                            (expectedTestResults, actualTestResults) = await filter.FilterResults(unmatchedResultTolerance, expectedTestResults, actualTestResults);
+                        });
+
                         return new CountingReportGenerator(
                             metadata.TestDescription,
                             trackingId,
                             metadata.ExpectedSource,
-                            expectedTestResults,
+                            expectedTestResults.GetAsyncEnumerator(),
                             metadata.ActualSource,
-                            actualTestResults,
+                            actualTestResults.GetAsyncEnumerator(),
```

Comment:
These generators took our custom enumerator type before, so we pass in the IAsyncEnumerator.